frontend/controller/serviceworkers/push.js

```diff
@@ -80,13 +80,15 @@ export default (sbp('sbp/selectors/register', {
       const socket = pubsub.socket
       const reported = map.get(socket)
       map.set(socket, subscriptionInfo)
-      if (subscriptionInfo?.endpoint && reported !== subscriptionInfo.endpoint) {
-        // If the subscription has changed, report it to the server
-        pubsub.socket.send(createMessage(
-          REQUEST_TYPE.PUSH_ACTION,
-          { action: PUSH_SERVER_ACTION_TYPE.STORE_SUBSCRIPTION, payload: subscriptionInfo }
-        ))
-      } else if (!subscriptionInfo && reported) {
+      if (subscriptionInfo?.endpoint) {
+        if (!reported || subscriptionInfo.endpoint !== reported.endpoint) {
```

Comment:
It looks like you've caught a bug here (`reported.endpoint`). I think that still the nested `if` can be avoided by using, e.g., `subscriptionInfo?.endpoint && reported?.endpoint !== subscriptionInfo.endpoint`, or by replacing the line above by `map.set(socket, subscriptionInfo?.endpoint)`

Reply:
I prefer the readability of this nested if, it's easier for me to follow what's going on as it reads more like English.

"If we have an endpoint, and either we did not have one before, or it's different, then report it".

Versus,

"If we have an endpoint and the one we had before or null is different from the one passed in, then report it."

It also then works better with the `else` clause below, as that can be read as:

"If we have an endpoint, or else if we have one from before"

Versus:

""If we have an endpoint and the one we had before or null is different from the one passed in, or else if we have one from before"